leetcode/src/19.c

```diff
@@ -0,0 +1,32 @@
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     struct ListNode *next;
+ * };
+ */
+
+struct ListNode *removeNthFromEnd(struct ListNode *head, int n) {
+    struct ListNode *p_free, *p = head;
+    int i, sz = 0;
+    while (p != NULL) {
+        p = p->next;
+        sz++;
+    }
```

Comment:
It would be more efficient if you use memoization, I checked the problem description, an array of 30 pointers should be plenty enough. That way you get to solve the follow up objective of only one pass

Reply:
Thanks for the comment, I agree that it is possible to solve the problem with memoization and I considered going this route, but decided to make algorithm more generic (allowing more than 30 nodes). Memoization approach would require to trade space for speed and I think would be an alternative solution all together, so happy to implement it at the later stage (or someone else will do it).